pkg/controller/tektoninstallation/tektoninstallation_controller_test.go

```diff
@@ -60,10 +84,122 @@ func TestTektonInstallationController(t *testing.T) {
 				Exists().
 				HasSpec(tektonSub.Spec)
 
+			AssertThatTektonInstallation(t, tektonInstallation.Namespace, tektonInstallation.Name, cl).
+				HasConditions(Unknown())
+		})
+
+	})
+
+	// reconciling on tektonconfig resource watcher
+	t.Run("tektonconfig watcher", func(t *testing.T) {
+
+		t.Run("installed tekton installation", func(t *testing.T) {
+			// given
+			tektonInstallation := NewInstallation()
+			installedCode := []config.ConfigCondition{
+				config.ConfigCondition{
+					Code: "applied-addons",
+				},
+				config.ConfigCondition{
+					Code: config.InstalledStatus,
+				},
+				config.ConfigCondition{
+					Code: "validated-pipeline",
+				},
+			}
+			tektonConfig := NewTektonConfig(installedCode...)
+			cl, r := configureClient(t, tektonInstallation,
+				NewSubscription(SubscriptionNamespace),
+				tektonConfig)
+			r.watchTektonConfig = func() error {
+				return nil
+			}
+			request := newReconcileRequest(tektonInstallation)
+
+			// when
+			_, err := r.Reconcile(request)
+
+			// then
+			require.NoError(t, err)
+			AssertThatSubscription(t, SubscriptionNamespace, SubscriptionName, cl).Exists()
 			AssertThatTektonInstallation(t, tektonInstallation.Namespace, tektonInstallation.Name, cl).
 				HasConditions(InstallationSucceeded())
 		})
 
+		t.Run("installing tekton installation", func(t *testing.T) {
+			// given
+			tektonInstallation := NewInstallation()
+			installingCode := config.ConfigCondition{
+				Code: config.InstallingStatus,
+			}
+			tektonConfig := NewTektonConfig(installingCode)
+			cl, r := configureClient(t, tektonInstallation,
+				NewSubscription(SubscriptionNamespace),
+				tektonConfig)
+			r.watchTektonConfig = func() error {
+				return nil
+			}
+			request := newReconcileRequest(tektonInstallation)
+
+			// when
+			_, err := r.Reconcile(request)
+
+			// then
+			require.NoError(t, err)
+			AssertThatSubscription(t, SubscriptionNamespace, SubscriptionName, cl).Exists()
+			AssertThatTektonInstallation(t, tektonInstallation.Namespace, tektonInstallation.Name, cl).
+				HasConditions(Installing("tekton installation installing: "))
```

Comment:
could we please verify the details here as well?

Reply:
@MatousJobanek did you mean the details we added here: https://github.com/codeready-toolchain/toolchain-operator/pull/67/files#diff-abc654745b2714d4bdd442f1a46078aeR137 or did you mean something to do with the spec?

Reply:
@MatousJobanek thanks for clarifying that you meant the details field. In order to do this, did we want to change the definition of `NewTektonConfig(...)` right now the func takes in a list of strings (as per the PR comments) but we could change this back to accepting a configuration so the details can be set... or we can hard code it in `NewTektonConfig` ... what do you think?

Reply:
as it's this only case, then I would set the details directly inside of the CodeCondition and keep the `NewTektonConfig` to take the list of strings